### PR TITLE
Include taskState in listLastFires

### LINKED ISSUE #6248: Hooks list fires "success" status might be misleading when task itself fails
Currently hook fire state "success" means that task was successfully created.

https://community-tc.services.mozilla.com/hooks/project-mozci/decision-task-production

image

We can add an extra column with the task status to show if it was successfull or not

### PR DESCRIPTION
**Pull request by lotas**
Hook's fire status is somewhat confusing to users as it make it seem as hook last fires were successful, while in fact it only represents "task was scheduled". 
To make it easier to spot issues earlier, `hooks.listLastFires` includes `taskState` column, which represents tasks' last run state at the time of the call. 
New column is also added in UI to show both hook fire status and the task state

`taskState` represents tasks' last run state at the time of the call.

Github Bug/Issue: Fixes #6248

image

### CODE REVIEW

File: services/hooks/src/api.js [hunk 1]
```diff
@@ -691,6 +691,7 @@ builder.declare({
       taskCreateTime: row.task_create_time.toJSON(),
       result: row.result,
       error: row.error,
+      taskState: row.task_state || 'unscheduled',
```
Comment (lotas) [inline services/hooks/src/api.js:694]: this is missing tests still, will add soon

**Review by matt-boris — CHANGES_REQUESTED** [review]
Changes look great! Just requesting the new db method to have an updated description. 👍🏻

File: db/versions/0088.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  end
+methods:
+  get_last_fires:
+    deprecated: true
+  get_last_fires_with_task_state:
+    description: |-
+      Get hooks last fires filtered by the `hook_group_id` and `hook_id` arguments,
+      ordered by `hook_group_id`, `hook_id`, and `task_create_time`.
+      If the pagination arguments are both NULL, all rows are returned.
+      Otherwise, page_size rows are returned at offset page_offset.
```
[the remark this answers is not in the document]
Comment (lotas) [reply db/versions/0088.yml:19]: right! thanks

**Review by matt-boris — APPROVED** [review]
Awesome, thanks Yarik! 🚢

**Pull request merged**